columnencoder.cpp

```diff
@@ -373,13 +373,65 @@ std::string ColumnEncoder::encodeRScript(std::string text, std::set<std::string>
 	return encodeRScript(text, encodingMap(), originalNames(), columnNamesFound);
 }
 
-std::string ColumnEncoder::encodeRScript(std::string text, const std::map<std::string, std::string> & map, const std::vector<std::string> & names, std::set<std::string> * columnNamesFound)
+/*!
+ * \brief Replace column names with encoded column names.
+ * \param allowedPrefixes Specifies all prefix that column names may have and still be seen as columns names. e.g. "data.<column_name>"
+ * \return Original string with column names replaced by encoded column names
+ */
+std::string ColumnEncoder::encodeRScript(std::string text, std::map<std::string, std::set<std::string>>& prefixedColumnsFound, const std::set<std::string>& allowedPrefixes)
+{
+	auto prefixes = allowedPrefixes;
```

Comment:
```suggestion
	stringvec prefixes = allowedPrefixes;
	std::sort(prefixes.begin(), prefixes.end(), [](const std::string & l, const std::string & r){ return l.size() < r.size();  }
```

This should make the order in which the prefixes are traversed a bit more predictable, and also make sure the non-prefixed ones are encoded first

Reply:
(which should make having both `V1` and `data.V1` in the data work together with this prefix thing)

Reply:
good catch

Reply:
Although @FBartos tells me he wants to check for this in the component already to disallow this anyway. So maybe it doesnt matter